test/uk/gov/hmrc/nisp/views/StatePension_MQPViewSpec.scala

```diff
@@ -0,0 +1,1206 @@
+    }
+    "render page with link  'View your National Insurence Record'" in {
+      assertEqualsMessage(htmlAccountDoc, "article.content__body>a:nth-child(8)", "nisp.main.showyourrecord")
+    }
+    "render page with href link  'View your National Insurence Record'" in {
+      assertLinkHasValue(htmlAccountDoc, "article.content__body>a:nth-child(8)", "/check-your-state-pension/account/nirecord")
+    }
+
+    /*overseas message*/
+    "render page with text  'As you are living or working overseas (opens in new tab), you may be entitled to a " +
+      "State Pension from the country you are living or working in.'" in {
+      assertEqualsMessage(htmlAccountDoc, "article.content__body>div.panel-indent:nth-child(9)>p", "nisp.main.overseas")
+    }
+    /*Ends*/
+
+    "render page with heading  'Putting of claiming'" in {
+      assertEqualsMessage(htmlAccountDoc, "article.content__body>h2:nth-child(10)", "nisp.main.puttingOff")
+    }
+
+    "render page with text  'When you are 67, you can put off claiming your State Pension. Doing this may mean you get extra State Pension when you do come to claim it. " +
+      "The extra amount, along with your State Pension, forms part of your taxable income.'" in {
+      assertContainsDynamicMessage(htmlAccountDoc, "article.content__body>p:nth-child(11)", "nisp.main.puttingOff.line1", "67", null, null)
+    }
+
+    "render page with link 'More on putting off claiming (opens in new tab)'" in {
+      assertEqualsMessage(htmlAccountDoc, "article.content__body>a:nth-child(12)", "nisp.main.puttingOff.linkTitle")
+    }
+    "render page with href link 'More on putting off claiming (opens in new tab)'" in {
+      assertLinkHasValue(htmlAccountDoc, "article.content__body>a:nth-child(12)", "https://www.gov.uk/deferring-state-pension")
+    }
+
+    /*Side bar help*/
+    "render page with heading  'Get help'" in {
+      assertEqualsMessage(htmlAccountDoc, "aside.sidebar >div.helpline-sidebar>h2", "nisp.nirecord.helpline.getHelp")
+    }
+    "render page with text  'Helpline 0345 608 0126'" in {
+      assertEqualsMessage(htmlAccountDoc, "aside.sidebar >div.helpline-sidebar>p:nth-child(2)", "nisp.nirecord.helpline.number")
+    }
+    "render page with text  'Monday to Friday: 8am to 6pm'" in {
+      assertEqualsMessage(htmlAccountDoc, "aside.sidebar >div.helpline-sidebar>p:nth-child(3)", "nisp.nirecord.helpline.openTimes")
+    }
+    "render page with text  'Calls cost up to 12p a minute from landlines. Calls from mobiles may cost more.'" in {
+      assertEqualsMessage(htmlAccountDoc, "aside.sidebar >div.helpline-sidebar>p:nth-child(4)", "nisp.nirecord.helpline.callsCost")
+    }
+    "render page with help text 'Get help with this page.' " in {
+      assertElementContainsText(htmlAccountDoc, "div.report-error>a#get-help-action", "Get help with this page.")
+    }
+
+  }
+
+  "Render State Pension view with MQP : Continue Working || Less than 10 years || Fill Gaps || Full Rate" should {
```

Comment:
@dspork  Restructured the unit tests. These scenarios and lines i have used from Acceptance tests so that its easy to make sure everything is covered scenario wise. :-)